api/config/v1alpha1/envoygateway_types.go

```diff
@@ -313,6 +320,35 @@ type ExtensionTLS struct {
 	CertificateRef gwapiv1b1.SecretObjectReference `json:"certificateRef"`
 }
 
+// EnvoyGatewayAdmin defines the Envoy Gateway Admin configuration.
+type EnvoyGatewayAdmin struct {
+
+	// Debug defines Debug Server configuration of Envoy Gateway
+	//
+	// +optional
+	Debug *DebugConfig `json:"debug,omitempty"`
+}
+
+// DebugConfig defines the Envoy Gateway Debug Server configuration.
+type DebugConfig struct {
+	// Enable defines if enable the debug server.
+	//
+	// +optional
+	// +kubebuilder:default=false
+	Enable bool
+	// Port defines the port the debug server is exposed on.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default=8899
+	Port int `json:"port,omitempty"`
```

Comment:
imo, `port` should live at the top level under `Admin` , because it is the port of the `Admin` Server. maybe even
a field like `Address` makes sense, in the future we can also can `IP`/`Host` in case we want to expose bind addr
```
admin:
  address:
    port: 8080
 ```